src/wallet/wallet.cpp

```diff
@@ -842,6 +842,25 @@ bool CWallet::IsSpentKey(const uint256& hash, unsigned int n) const
     return false;
 }
 
+void CWallet::SetUsedAddressState(WalletBatch& batch, const CTxOut& cout, bool used)
+{
+    AssertLockHeld(cs_wallet);
```

Comment:
I'm getting compiler warnings consistent with the CI:

```
wallet/wallet.cpp:847:5: warning: calling function 'AssertLockHeldInternal<AnnotatedMixin<std::__1::recursive_mutex> >' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
    AssertLockHeld(cs_wallet);
    ^
./sync.h:81:28: note: expanded from macro 'AssertLockHeld'
#define AssertLockHeld(cs) AssertLockHeldInternal(#cs, __FILE__, __LINE__, &cs)
                           ^
wallet/wallet.cpp:849:55: warning: calling function 'IsMine' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
    if (ExtractDestination(cout.scriptPubKey, dst) && IsMine(dst)) {
                                                      ^
wallet/wallet.cpp:850:22: warning: calling function 'GetDestData' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
        if (used && !GetDestData(dst, "address_used", nullptr)) {
                     ^
wallet/wallet.cpp:851:13: warning: calling function 'AddDestData' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
            AddDestData(batch, dst, "address_used", "p"); // p for "present", opposite of absent (null)
            ^
wallet/wallet.cpp:852:29: warning: calling function 'GetDestData' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
        } else if (!used && GetDestData(dst, "address_used", nullptr)) {
                            ^
wallet/wallet.cpp:853:13: warning: calling function 'EraseDestData' requires holding mutex 'cs_wallet' exclusively [-Wthread-safety-analysis]
            EraseDestData(batch, dst, "address_used");
            ^
6 warnings generated.
```

Reply:
Thanks for taking a look @adamjonas, this should now be resolved in latest commit